sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -351,7 +351,11 @@ def truncate_attribute_values(attrs)
           return EMPTY_ATTRIBUTES if attrs.nil?
 
           attribute_length_limit = @span_limits.attribute_length_limit
-          attrs.each { |key, value| attrs[key] = OpenTelemetry::Common::Utilities.truncate(value, attribute_length_limit) } if attribute_length_limit
+          return attrs if attribute_length_limit.nil?
+
+          attrs.each do |key, value|
+            attrs[key] = OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) if value.is_a?(String)
+          end
```

Comment:
And, in fact, we may as well simplify this to:
```suggestion
          attrs.transform_values! { |value| OpenTelemetry::Common::Utilities.truncate_attribute_value(value, attribute_length_limit) }
```

Reply:
Nice, but doesn't it still need the ` if value.is_a?(String)`?

Reply:
No, because that'll miss the `Array` case, and there's diminishing returns to `if value.is_a?(String) || value.is_a?(Array)` over just dealing with it in `truncate_attribute_value`.

Reply:
Hehe, like we already discussed and I mixed up :)